opendcs-rest-api/src/main/java/org/opendcs/odcsapi/res/SiteResources.java

```diff
@@ -30,99 +36,233 @@
+			}
+			siteRef.setPublicName(site.getPublicName());
+			siteRef.setDescription(site.getDescription());
+			HashMap<String, String> siteNames = new HashMap<>();
+			for(Iterator<SiteName> iter = site.getNames(); iter.hasNext(); )
+			{
+				final SiteName sn = iter.next();
+				siteNames.put(sn.getNameType(), sn.getNameValue());
+			}
+			siteRef.setSitenames(siteNames);
+			retList.add(siteRef);
+		}
+		return retList;
+	}
+
 	@GET
 	@Path("site")
 	@Produces(MediaType.APPLICATION_JSON)
 	@RolesAllowed({AuthorizationCheck.ODCS_API_GUEST})
-	public Response geSiteFull(@QueryParam("siteid") Long siteId)
-		throws WebAppException, DbException
+	public Response getSiteFull(@QueryParam("siteid") Long siteId)
+			throws WebAppException, DbException
 	{
 		if (siteId == null)
-			throw new WebAppException(ErrorCodes.MISSING_ID, 
-				"Missing required siteid parameter.");
+		{
+			throw new WebAppException(ErrorCodes.MISSING_ID,
+					"Missing required siteid parameter.");
+		}
 
-		Logger.getLogger(ApiConstants.loggerName).fine("getSite id=" + siteId);
-		try (DbInterface dbi = new DbInterface();
-			ApiSiteDAO dao = new ApiSiteDAO(dbi))
+		try (SiteDAI dai = getLegacyTimeseriesDB().makeSiteDAO();
+			 PropertiesDAI propsDai = getLegacyTimeseriesDB().makePropertiesDAO())
 		{
-			return ApiHttpUtil.createResponse(dao.getSite(siteId));
+			DbKey siteKey = DbKey.createDbKey(siteId);
+			Site returnedSite = dai.getSiteById(siteKey);
+			Properties props = new Properties();
+			propsDai.readProperties("SITE_PROPERTY", "SITE_ID", siteKey, props);
+			return Response.status(HttpServletResponse.SC_OK)
+					.entity(map(returnedSite, props)).build();
+		}
+		catch (NoSuchObjectException e)
+		{
+			return Response.status(HttpServletResponse.SC_NOT_FOUND)
+					.entity("Requested app with matching ID not found").build();
+		}
+		catch(DbIoException e)
+		{
+			throw new DbException("Unable to retrieve site by ID", e);
 		}
 	}
 
+	static ApiSite map(Site site, Properties properties)
+	{
+		ApiSite returnSite = new ApiSite();
+		if (site.getId() != null)
+		{
+			returnSite.setSiteId(site.getId().getValue());
+		}
+		else
+		{
+			returnSite.setSiteId(DbKey.NullKey.getValue());
+		}
+		returnSite.setLocationtype(site.getLocationType());
+		returnSite.setElevation(site.getElevation());
+		returnSite.setElevUnits(site.getElevationUnits());
+		returnSite.setActive(site.isActive());
+		returnSite.setDescription(site.getDescription());
+		returnSite.setLastModified(site.getLastModifyTime());
+		returnSite.setCountry(site.country);
+		returnSite.setState(site.state);
+		returnSite.setProperties(properties);
+		returnSite.setNearestCity(site.nearestCity);
+		returnSite.setLatitude(site.latitude);
+		returnSite.setLongitude(site.longitude);
+		returnSite.setTimezone(site.timeZoneAbbr);
+		returnSite.setRegion(site.region);
+		returnSite.setPublicName(site.getPublicName());
+		HashMap<String, String> siteNames = new HashMap<>();
```

Comment:
I'd think this would be Map<String, String> and the impl class LinkedHashMap

Reply:
I updated the impl class, but the ApiSiteRef DTO requires a HashMap object containing the sitenames.

Reply:
Is there a reason ApiSiteRef requires HashMap?  Thats an implementation, methods should require and return interfaces anywhere thats possible.